genesyscloud/sdk_client_pool.go

```diff
@@ -44,21 +44,41 @@ func InitSDKClientPool(max int, version string, providerConfig *schema.ResourceD
 }
 
 func (p *SDKClientPool) preFill(providerConfig *schema.ResourceData, version string) diag.Diagnostics {
-	for cap(p.pool) > 0 {
+	errorChan := make(chan diag.Diagnostics)
+	wgDone := make(chan bool)
+	var wg sync.WaitGroup
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	for i := 0; i < cap(p.pool); i++ {
 		sdkConfig := platformclientv2.NewConfiguration()
-		err := initClientConfig(providerConfig, version, sdkConfig)
-		if err != nil {
-			return err
-		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err := initClientConfig(providerConfig, version, sdkConfig)
```

Comment:
What are you doing here Ronan?  Is this taking a pre-configured list of clientConfigs from a pool?  If it is, Nice?

Reply:
Basically it's got a channel with a list of SDK configs and it's authenticating each one. My updates to this code are to make it multi threaded to speed it up.